formula1_race_analysis/q1_report_generator.py

```diff
@@ -0,0 +1,50 @@
+import sys
+from pathlib import Path
+
+import click
+import structlog
+
+from formula1_race_analysis.custom_exceptions import CustomFileNotFoundError
+from formula1_race_analysis.logging_config import logger_factory
+from formula1_race_analysis.q1_session_analyzer import display_race_report, get_results_of_first_qualifying_session
+
+logger_factory()
+logger: structlog.stdlib.BoundLogger = structlog.get_logger()
+
+
+@click.command()
+@click.option("--data_dir", type=Path, required=True, help="[--asc | --des]")
+@click.option("--asc", "order", flag_value="asc", default=True, help="To sort results in ascending order")
+@click.option("--des", "order", flag_value="des", default=False, help="To sort results in descending order")
+@click.option("--driver", default=None)
+def generate_report(data_dir: str, order: str, driver: str) -> None:
+    if not data_dir or not Path(data_dir).is_dir():
```

Comment:
I think these checks are redundant, data_dir is required click option, it has type Path so Path(..) is redundant. Only check is_dir needed